core/types/hashing_test.go

```diff
@@ -38,7 +39,8 @@ func TestDeriveSha(t *testing.T) {
 		t.Fatal(err)
 	}
 	for len(txs) < 1000 {
-		exp := types.DeriveSha(txs, new(trie.Trie))
+		tr, _ := trie.New(common.Hash{}, trie.NewDatabase(rawdb.NewMemoryDatabase()))
+		exp := types.DeriveSha(txs, tr)
```

Comment:
Why did you make these changes? Old code seems to work fine, without instantiating new databases

Reply:
The original initialization `new(trie.Trie)` will create the trie structure but leave all pointers as nil.
Right now it's `db` and `tracer`. Although in this case both of them are not used, but it's still very dangerous to leave pointers as nil.

I did a lots of this kind changes to ensure all the created trie instances are initialized properly.